core/src/main/clojure/xtdb/tx_producer.clj

```diff
@@ -24,7 +24,7 @@
 
 (s/def :xt/id any?)
 (s/def ::doc (s/and (s/keys :req [:xt/id])
-                    (s/conformer #(update-keys % util/ns-kw->kw) #(update-keys % util/kw->ns-kw))))
+                    (s/conformer #(update-keys % util/kw->normal-form-kw) #(update-keys % util/normal-form-kw->datalog-form-kw))))
```

Comment:
Question is if this conforming should also happen in the indexer?

Reply:
I would guess so - we'll probably want to normalise the SQL cols in insert/update/etc, just in case a SQL user gives us an unnormalised key

Reply:
@jarohen Yes already doing that. I was more wondering if it's ok to have "insertion" normalization happening at two different places. One on spec conform (Datalog `:put`) and one in the indexer (`:insert` + `:update`).